File: blur-search/modules/background-blur/android/src/main/java/expo/modules/backgroundblur/BackgroundBlurModule.kt
package expo.modules.backgroundblur

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class BackgroundBlurModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("BackgroundBlur")

    View(BackgroundBlurView::class) {
      Prop("radius") { view: BackgroundBlurView, value: Double ->
        view.setRadius(value)
      }

      Prop("tintColor") { view: BackgroundBlurView, value: String? ->
        view.setTintColor(value)
      }

      Prop("tintOpacity") { view: BackgroundBlurView, value: Double ->
        view.setTintOpacity(value)
      }
    }
  }
}

File: blur-search/modules/background-blur/ios/BackgroundBlurModule.swift
import ExpoModulesCore

public class BackgroundBlurModule: Module {
  public func definition() -> ModuleDefinition {
    Name("BackgroundBlur")

    View(BackgroundBlurView.self) {
      Prop("radius") { (view: BackgroundBlurView, value: Double) in
        view.setRadius(value)
      }

      Prop("tintColor") { (view: BackgroundBlurView, value: String?) in
        view.setTintColor(value)
      }

      Prop("tintOpacity") { (view: BackgroundBlurView, value: Double) in
        view.setTintOpacity(value)
      }
    }
  }
}
